model/contact/contacts.go

```diff
@@ -139,12 +139,19 @@ func (c *Contact) PrimaryCozyURL() string {
 	return url
 }
 
-// FillFullnameIfMissing can be used to add a fullname if there was none.
-func (c *Contact) FillFullnameIfMissing(fullname string) {
-	name, ok := c.Get("fullname").(string)
-	if !ok || len(name) == 0 {
-		c.M["fullname"] = fullname
-	}
+// AddNameIfMissing can be used to add a name if there was none.
+func (c *Contact) AddNameIfMissing(db prefixer.Prefixer, name string) error {
+	was, ok := c.Get("displayName").(string)
+	if ok && len(was) > 0 {
+		return nil
```

Comment:
Are we sure we'll always have a `fullname` if `displayName` is present? Otherwise, this early return will prevent filling in the missing `fullname`.

Reply:
No, but I would prefer to avoid filling a `fullname` if `displayName` is present, as it is really unclear what would be the correct way to fill it.

Reply:
So, doesn't that mean we only really care about the `displayName` and we should always avoid filling in the `fullname`?

Reply:
No, I think we should fill `fullname` if `displayName` is missing. As I understand, a contact can have a `fullname`, a first name, a last name, a pseudo, etc. And `displayName` is a way to take the name that is the more meaningful between all the variants. It is persisted in CouchDB and not computed at runtime to allow indexing it and making requests sorting by it.

Reply:
OK. I still find a bit weird to have a display name without a full name but if we don't have the data, I guess there's nothing we can do about it.